proof/invariant-abstract/DetSchedSchedule_AI.thy

```diff
@@ -9459,33 +9495,41 @@ lemma refill_budget_check_round_robin_sc_refills_update_unspecified:
   done
 
 lemma commit_time_released_ipc_queues[wp]:
-  "\<lbrace>released_ipc_queues\<rbrace>
+  "\<lbrace>released_ipc_queues
+    and cur_sc_more_than_ready
+    and active_sc_valid_refills
+    and current_time_bounded 0\<rbrace>
    commit_time
    \<lbrace>\<lambda>_. released_ipc_queues\<rbrace>"
   supply if_split[split del]
   apply (simp add: commit_time_def)
   apply (rule hoare_seq_ext[OF _ gets_sp])
-  apply (rule hoare_seq_ext[OF _ gets_sp])
   apply (rule hoare_seq_ext[OF _ get_sched_context_sp])
   apply (rule hoare_seq_ext[where B="\<lambda>_. released_ipc_queues"], wpsimp)
   apply (rule hoare_when_cases, simp)
+  apply (rule hoare_seq_ext[OF _ gets_sp])
+  apply (rename_tac csc sc consumed)
   apply (rule hoare_seq_ext[where B="\<lambda>_. released_ipc_queues"], wpsimp)
```

Comment:
I haven't checked in Isabelle, but if you're going to have this `hoare_seq_ext` with `B` just equal to the original postcondition, do you want to instead change the `gets_sp` line before to a regular `hoare_seq_ext_skip`? You should end up in the same spot. Minor though, of course

Reply:
In this case I'm using `hoare_seq_ext` to skip the last command, not the first. So the gets_sp commands are still needed because the precondition is maintained. But I've added `hoare_seq_ext_skip'` to facilitate skipping the last command, and that's a lot neater!